planning/behavior_path_planner/src/utils/start_planner/shift_pull_out.cpp

```diff
@@ -155,17 +155,26 @@ std::vector<PullOutPath> ShiftPullOut::calcPullOutPaths(
     start_planner_utils::calcEndArcLength(s_start, forward_path_length, road_lanes, goal_pose);
   const double s_end = path_end_info.first;
   const bool path_terminal_is_goal = path_end_info.second;
-
   constexpr double RESAMPLE_INTERVAL = 1.0;
   PathWithLaneId road_lane_reference_path = utils::resamplePathWithSpline(
     route_handler.getCenterLinePath(road_lanes, s_start, s_end), RESAMPLE_INTERVAL);
+  const double shift_distance = s_end - s_start;
+  const size_t shift_end_idx =
+    findNearestIndex(road_lane_reference_path.points, goal_pose.position);
```

Comment:
goal_pose is from the route and is not the same to shift end pose.
end_pose is `shift_end_pose_ptr` and the idx is `shift_line_idx.end_idx`

```
    // create shift line
    ShiftLine shift_line{};
    shift_line.start = start_pose;
    shift_line.end = *shift_end_pose_ptr;
    shift_line.end_shift_length = shift_length;
    path_shifter.addShiftLine(shift_line);
    path_shifter.setVelocity(0.0);  // initial velocity is 0
    path_shifter.setLongitudinalAcceleration(longitudinal_acc);
    path_shifter.setLateralAccelerationLimit(lateral_acc);

    const auto shift_line_idx = path_shifter.getShiftLines().front();
    if (!has_non_shifted_path && (shift_line_idx.end_idx - shift_line_idx.start_idx <= 1)) {
      candidate_paths.push_back(non_shifted_path);
      has_non_shifted_path = true;
      continue;
    }
```